src/type.cpp

```diff
@@ -34,10 +43,39 @@ bool PtrType::IsEqual(const Type& that) const noexcept {
   return false;
 }
 
+std::size_t PtrType::ToSize() const {
+  return size_;
+}
+
 std::string PtrType::ToString() const {
   return base_type_->ToString() + "*";
 }
 
 std::unique_ptr<Type> PtrType::Clone() const {
   return std::make_unique<PtrType>(base_type_->Clone());
 }
+
+bool ArrType::IsEqual(const Type& that) const noexcept {
+  if (const auto* that_arr = dynamic_cast<const ArrType*>(&that)) {
+    return that_arr->base_type_->IsEqual(*base_type_);
+  } else if (const auto* that_prim = dynamic_cast<const PrimType*>(&that)) {
+    return that_prim->IsEqual(*base_type_);
+  }
+  return false;
+}
```

Comment:
I think we need some revision on this. As mentioned in the reference manual at section 6.7.5.2:

- For two array types to be **compatible**, both must have compatible element types. Additionally, if both size specifiers are present and are integer constant expressions, then both size specifiers must have the same constant value.

The term **compatible** in the manual is akin to **equal** in our implementation.